### PR TITLE
Add GCP Region Names API

### PR DESCRIPTION
**Pull request by noahl**
A simple API to pull out the parts of a GCP zone, region, or
multi-region location name.

### CODE REVIEW

**Comment by noahl** [comment]
@stfeng @skschneider @cmccoy please take a look.

This is part of a plan to address #837, and also to let users select the multi-region location to test GCS in. This PR provides a general API for pulling out parts of a GCP location name. Once we have this, I can have the `object_storage_service` benchmark handle different types of location.

**Comment by stfeng** [comment]
I don't think we need this class to address 837 for object storage
benchmark. All we need is a new storage_continent flag to allow user to
choose a continental location.

We should not try to infer object storage locations from *gce* zones, these
two are different concepts.

File: perfkitbenchmarker/providers/gcp/util.py [hunk 1]
```diff
@@ -113,3 +113,61 @@ def _AddCommonFlags(self, resource):
     if hasattr(resource, 'zone'):
       self.flags['zone'] = resource.zone
     self.additional_flags.extend(FLAGS.additional_gcloud_flags or ())
+
+
+class GCPLocation(object):
+  """A GCP location.
+
+  This class may represent a zone, region, or continent.
```
Comment (cmccoy) [inline perfkitbenchmarker/providers/gcp/util.py]: I don't see continent anywhere...

**Comment by stfeng** [comment]
If we want to have a generic way of getting region info from a GCP "zone string" (such as "us-central1-a"), we should explore ways to use official google APIs first if possible.

Anyway, it's an interesting thought to make this a library, but I don't see an immediate use case for it (see my other comments above)

File: perfkitbenchmarker/providers/gcp/util.py (hunk 1 above)
Comment (noahl) [inline perfkitbenchmarker/providers/gcp/util.py]: Oops, left over from an old version. "Multiregions" were initially "continents" until I renamed them. Looks like I missed this reference.

**Comment by noahl** [comment]
@stfeng The reason I wanted an API like this is that we could make the object storage benchmark run the continental test in the continent corresponding to the region passed in by the `object_storage_region` flag, which I think would be logical (I should have done it when I added the region flag).

Also, it looks to me from https://cloud.google.com/docs/geography-and-regions like all GCP products have the same zones and regions. I think it would actually be useful to connect the two, so we could distinguish when a GCS benchmark is using VMs in a zone that is part of the bucket's region versus a zone that is not part of the bucket's region.

**Comment by stfeng** [comment]
My point is we can introduce a separate flag to specify which continent to use for object storage test, e.g. "object_storage_continent", instead of overloading the object_storage_region flag, it's much simpler and more specific this way, and it's also more extensible because it allows the user to specify provider-specific continent strings, instead of implying it from regional name, which may be different. e.g., suppose in the future when a provider introduce a continental offering that is named "US_2", or ASIA_B, then it would be quite hard to jam it into the object_storage_region flag.

**Comment by noahl** [comment]
@stfeng I see what you're saying, and I need to rethink the design of this. Let's talk in person.

**Pull request closed without merging**